lib/require_all.rb

```diff
@@ -5,6 +5,8 @@
 #++
 
 module RequireAll
+  RequireError = Class.new(StandardError)
```

Comment:
Why not inherit from NameError or StandardError instead of going that high up in hierarchy?

Reply:
> Why not inherit from NameError or StandardError instead of going that high up in hierarchy?

This exception _does_ inherit from `StandardError` - not sure what you mean? It could inherit from `NameError`, I guess? I'm not sure that it [_is a_](https://en.wikipedia.org/wiki/Liskov_substitution_principle) `NameError`, rather than 'can be caused by one'.

Re name - how about just `RequireAllError`?

Reply:
Looking again at it - maybe this error should inherit from `LoadError` instead so that there would be less chance of any existing code accidentally rescue'ing this error (StandardError is catched when rescue-ing without specifying any explicit exception class for example).

Then it makes more sense to have `RequireAll::LoadError` as its name and that's what this error basically is all about.